client.go

```diff
@@ -147,31 +160,42 @@ func (c *Client) httpPut(url string, bodyType string, body io.Reader) (
 	if err != nil {
 		return nil, err
 	}
-	return c.httpClient.Do(req)
+
+	res, err := c.httpClient.Do(req)
+	log.Debugln(loggerResponsePrepend, logging.HTTPMessageLogger(res))
+	return res, err
 }
 
 func (c *Client) httpGet(url string) (*http.Response, error) {
 	req, err := http.NewRequest("GET", url, nil)
+	log.Debugln(loggerRequestPrepend, logging.HTTPMessageLogger(req))
 	if err != nil {
 		return nil, err
 	}
 	err = c.auth.AddAuthentication(req)
 	if err != nil {
 		return nil, err
 	}
-	return c.httpClient.Do(req)
+
+	res, err := c.httpClient.Do(req)
+	log.Debugln(loggerResponsePrepend, logging.HTTPMessageLogger(res))
+	return res, err
 }
 
 func (c *Client) httpDelete(url string) (*http.Response, error) {
 	req, err := http.NewRequest("DELETE", url, nil)
+	log.Debugln(loggerRequestPrepend, logging.HTTPMessageLogger(req))
 	if err != nil {
 		return nil, err
 	}
 	err = c.auth.AddAuthentication(req)
 	if err != nil {
 		return nil, err
 	}
-	return c.httpClient.Do(req)
+
+	res, err := c.httpClient.Do(req)
+	log.Debugln(loggerResponsePrepend, logging.HTTPMessageLogger(res))
```

Comment:
Now I'm wondering, whether we should log the request BEFORE calling httpClient.Do, and log the response AFTER we call httpClient.Do. Because http.Do might error out and we will never see the offending request.

To achieve this, we could wrap http.Client.Do into some function, like

```
httpSend(req) err {
  logRequest
  res, err = httpClient.Do(req)
  logResponse // Not sure if we should log it if err happened.
  return err
}
```